viewing_party/party.py

```diff
@@ -1,23 +1,146 @@
+import statistics
 # ------------- WAVE 1 --------------------
+user_data = {}
 
-def create_movie(title, genre, rating):
-    pass
+def create_movie(movie_title, genre, rating):
+    if movie_title and genre and rating:
+        new_movie = {
+            "title": movie_title,
+            "genre": genre,
+            "rating": rating
+        }
+        return new_movie
+    else:
+        return None
 
+def add_to_watched(user_data, movie):
+    user_data["watched"]= [movie]
+    return user_data
+
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"]= [movie]
+    return user_data
+
+def watch_movie(user_data, movie):
+    for movie_dict in user_data["watchlist"]:
+        if movie_dict["title"] == movie:
+            user_data["watched"].append(movie_dict)
+            user_data["watchlist"].remove(movie_dict)   
+    return user_data
+    
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+    rating_list = [movie_dict["rating"]for movie_dict in user_data["watched"]]
+    if rating_list != []:
+        average = sum(rating_list)/len(user_data["watched"])
+    else:
+        average = 0.0
+    return average
 
+def get_most_watched_genre(user_data):
+    genre_list = [movie_dict["genre"] for movie_dict in user_data["watched"]]
+
+    if genre_list != []:
+        popular_genre = statistics.mode(genre_list)
+        return popular_genre
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
-# -----------------------------------------
+# ----------------------------------------
+
+def get_unique_watched(user_data):
+    # lists of movies watched by user, and by friends 
+    movies_user_watched = []
+    movies_friends_watched = []
+
+    # add movie titles to the lists
+    for i in range(len(user_data["watched"])):
+        movie_title = user_data["watched"][i]['title']
+        movies_user_watched.append(movie_title)
+
+    for i in range(len(user_data["friends"])):
+        list_of_dicts = user_data["friends"][i]['watched']
+        for i in range(len(list_of_dicts)):
+            movies_friends_watched.append(list_of_dicts[i]['title'])
+    
+    # use set to compare the lists, taking the difference between the user's set and the friends set.
+    unique_user_set = set(movies_user_watched) - set(movies_friends_watched)

```

Comment:
For a slight space optimization I recommend building the sets on lines 62 and 67, rather than building lists and then converting them to sets.